src/v/redpanda/debug_bundle.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "seastarx.h"
+#include "utils/request_auth.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/core/iostream.hh>
+#include <seastar/core/sharded.hh>
+#include <seastar/core/sstring.hh>
+#include <seastar/http/file_handler.hh>
+#include <seastar/http/reply.hh>
+#include <seastar/http/request.hh>
+
+#include <memory>
+#include <optional>
+
+static constexpr ss::shard_id debug_bundle_shard_id = 0;
+
+struct debug_bundle_params {
+    std::optional<ss::sstring> logs_since;
+    std::optional<ss::sstring> logs_until;
+    std::optional<ss::sstring> logs_size_limit;
+    std::optional<ss::sstring> metrics_interval;
+
+    debug_bundle_params()
+      : logs_since{std::nullopt}
+      , logs_until{std::nullopt}
+      , logs_size_limit{std::nullopt}
+      , metrics_interval{std::nullopt} {}
+};
+
+enum class debug_bundle_status { initialized, running, complete, deleted };
+
+class debug_bundle : public ss::peering_sharded_service<debug_bundle> {
```

Comment:
are there per-core data? i would have assumed that every request would be handled on a single core, in which case, you could create a singleton service.

Reply:
I may be misunderstanding what a singleton service is (I assume it's a `std::unique_ptr`) is that the bundle is managed on the Admin server which is sharded. So, within the admin server, I would have to make sure everything runs on `debug_bundle_shard_id` regardless. Therefore, I figured wrapping everything in a service class would be easier.

Reply:
>may be misunderstanding what a singleton service is (I assume it's a std::unique_ptr)

yes

> but the problem I faced with this is that the bundle is managed on the Admin server which is sharded. 

right, it's a little bit awkward the way the admin_server is setup to put a singleton in here. i think you can ignore my comment, and go with your version. it isn't hurting anything and it's still efficient.